platforms/web/src/hooks/useBootstrapApp.ts

```diff
@@ -17,13 +20,30 @@ type Resources = {
 };
 
 export const useBootstrapApp = (onReady: () => void) => {
-  const { data, isLoading, error, isSuccess, refetch } = useQuery<Resources, AppError>('config-init', applicationController.initializeApp, {
-    refetchInterval: false,
-    retry: 1,
-    onSuccess: onReady,
-    cacheTime: CACHE_TIME,
-    staleTime: STALE_TIME,
-  });
+  const analyticsLoadedRef = useRef(false);
+  const { data, isLoading, error, isSuccess, refetch } = useQuery<Resources, AppError>(
+    'config-init',
+    () => applicationController.initializeApp(window.location.href),
+    {
+      refetchInterval: false,
+      retry: 1,
+      onSettled: async (data) => {
+        // because theming and analytics are specific to web, we configure it from here
+        // alternatively, we can use an events or specific callbacks or extend the AppController for each platform
+        if (data?.config) {
+          setThemingVariables(data?.config);
+        }
+
+        if (data?.config.analyticsToken && !analyticsLoadedRef.current) {
+          await addScript('/jwpltx.js');
+          analyticsLoadedRef.current = true;
+        }
+      },
```

Comment:
I see your comment, but I'm not so sure about this web-specific code. Since `useBootstrapApp()` is called from the Root, perhaps we can implement the onSettled callback immediately?

Reply:
I'm not sure if I understand what you mean. The `useBootstrapApp` also lives in the `platforms/web` package. Because I don't have a solution yet for web-specific code, I decided that all of this should live in the `platforms/web` package. 

Refactoring this using the `onSettled` callback only moves the handler from the hook to the container, right? Not sure what we gain by doing so.

Reply:
Sorry, I overlooked that this whole hook lives in `/web`. But perhaps it should move to `hooks-react` and keep only the `onSettled` handler in `/web/src/containers/Root`? That's what I meant actually.

Reply:
Pushed a change, what do you think?

Reply:
I like it, but one suggestion, see comment